lib/adal-angular.js

```diff
@@ -197,9 +197,35 @@
                     return false;
                 }
 
-                var routeChangeHandler = function (e, nextRoute) {
-                    var nextRouteUrl;
+                function getStates(toState) {
+                    var state = null;
+                    var states = [];
+                    if (toState.hasOwnProperty('parent')) {
+                        state = toState;
+                        while (state) {
+                            states.unshift(state);
+                            state = $injector.get('$state').get(state.parent);
+                        }
+                    }
+                    else {
+                        var stateNames = toState.name.split('.');
+                        var stateName = stateNames[0];
+                        state = $injector.get('$state').get(stateName);
+                        if (state) {
+                            states.push(state);
+                        }
+                        for (var i = 1; i < stateNames.length; i++) {
+                            stateName += '.' + stateNames[i];
+                            state = $injector.get('$state').get(stateName);
+                            if (state) {
+                                states.push(state);
+                            }
+                        }
```

Comment:
You can do something like this here to reduce code redundancy:

                       for (var i = 0, stateName = stateNames[0]; i < stateNames.length; i++) {
                            state = $injector.get('$state').get(stateName);
                            if (state) {
                                states.push(state);
                            }
                            stateName += '.' + stateNames[i];
                        }